### DISCUSSION TITLE
Frozen during firmware programming

### DISCUSSION THREAD

**Discussion by dlxmax** [tjboldt/Apple2-IO-RPi#146] [General]
I have some AliExpress AT28C64B's here. Four of them stop instantly at "writing page 0 to slot 7" with two rows of lines underneath. The other leads to a weird graphics crash at boot (I guess it's not a blank chip).

I had to make some substitutions. Perhaps that's the problem?

74HCT245 instead of 74LVC245 (also AliExpress)
74HC32 instead of 74LS32

Otherwise, the soldering is very good. All chips are in the right place and orientation. A 74LS245 is in right spot.
Thanks for the help.

**Comment by ozw1z5rd** [comment]
Hola, Do you know that the Write Cycle Time can't exceed 10 ms on the AT28C64B? 
Better you wait at least 10ms before doing something else.

74LVC245 are perfect, don't replace them. 
Replacing 74LS32 with 74HC32 do no effect.

**Comment by dlxmax** [comment]
I certainly waited more than 10ms!
Also, I used 74HCT245 and not 74LVC245.

**Reply by ozw1z5rd** [reply]
To debug such a problem, you should consider to remove the 245 that connected the Data bus, looking at the schematic in PDF is U0. Then try some random RW access ( of course you should get garbage ) if anything does not freeze, focus on U2 and try to replace it.  Check for bad solders.

**Reply by dlxmax** [reply]
I think you've misunderstood the situation. I cannot complete initial firmware programming.

**Reply by ozw1z5rd** [reply]
I understood it, and I’m suggesting to verify that the part of the circuit is working as expected.

**Comment by dlxmax** [comment]
@tjboldt I hope you can advise if my choice of alternative chips is the issue. Thanks.

**Reply by tjboldt** [reply]
The 74HCT245 requires a minimum 4.5V VCC but the board is wired to protect the RPi and is only supplying 3.3V to the 74LVC245. The HCT will not work as a replacement part for the LVC.

**Reply by dlxmax** [reply]
@tjboldt Thanks for the help. Unfortunately, even with 74LVC245's the error is the same.
Might you have any other ideas?

**Reply by Hubidrei** [reply]
74HC245 or better 74AHC245 work also here. I have tested them both, now my card works with AHC245 for a few months and never had any issues.

**Comment by Hubidrei** [comment]
> The 74HCT245 requires a minimum 4.5V VCC but the board is wired to protect the RPi and is only supplying 3.3V to the 74LVC245. The HCT will not work as a replacement part for the LVC.

**Comment by Hubidrei** [comment]
Did you test the EEPROMs in an external programmer?
I did also buy 5pcs on AliExpress, tested them in a "willem programmer" clone-, result: 2 of 5 did not work correctly.
